### PR TITLE
Basic obstacle avoidance for #US1

### PR DESCRIPTION
**Pull request by oodam**
Contributor @Novenius

### CODE REVIEW

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -1,9 +1,76 @@
...
+        }
+        
+        
+    }
+   
 }
+
+void autoStop(){
+  
+  if(front.getDistance() < 1000 && front.getDistance() > 0){
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:62]: Isn't `1000` too much? The distance is in centimeters.

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -1,9 +1,76 @@
...
+    }
+   
 }
+
+void autoStop(){
+  
+  if(front.getDistance() < 1000 && front.getDistance() > 0){ 
+                car.setSpeed(-50);
+                delay(3000);
+                car.setSpeed(0);
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:65]: Why is this needed? Doesn't the car stop quick enough due to inertia?

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (Novenius) [reply arduino/smartcar/smartcar.ino:62]: This must be a typo, it should be a 100

Comment (platisd) [reply arduino/smartcar/smartcar.ino:62]: After the value is corrected, there's another issue. `getDistance` is called twice meaning that there are two _different_ measurements taking place. What happens if the sensor returns different values, e.g. 0 and 200?

**Review by platisd — CHANGES_REQUESTED** [review]

File: arduino/smartcar/smartcar.ino (hunk 2 above)
Comment (Novenius) [reply arduino/smartcar/smartcar.ino:65]: If we only have:
if(front.getDistance() < 100 && front.getDistance() > 0){ 
car.setSpeed (0);
}

then the car will stop when detecting an object. Furthermore, if the object is static (like a mountain or a tree) the distance won't change and the car speed will constantly be set to 0. So we added that the car is put in reverse for 3000 milliseconds and then breaks so that the front.getDistance() > 100, which will allow for new commands to be received.

Comment (platisd) [reply arduino/smartcar/smartcar.ino:65]: Aha I see. Not how I would do it (I would just not allow the car to move forward but it would be OK to move backward) but I guess you will eventually figure out a way on how to do this as the project evolves.

Generally having such large delays in your execution can prove very problematic.

**Review by platisd — APPROVED** [review]
